### PR TITLE
KAFKA-13078: Closing FileRawSnapshotWriter too early

### PR DESCRIPTION
**Pull request by jsancio**
FollowerState state should close the store fetchingSnapshot if it exists
instead of the new snapshot begin set.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: raft/src/main/java/org/apache/kafka/raft/FollowerState.java [hunk 1]
```diff
@@ -137,11 +137,11 @@ public boolean updateHighWatermark(OptionalLong highWatermark) {
         return fetchingSnapshot;
     }
 
-    public void setFetchingSnapshot(Optional<RawSnapshotWriter> fetchingSnapshot) {
+    public void setFetchingSnapshot(Optional<RawSnapshotWriter> newSnapshot) {
```
Comment (jsancio) [inline raft/src/main/java/org/apache/kafka/raft/FollowerState.java:140]: Verified that a few of the tests failed without this and the changes to `MockRawSnapshotWriter` in this PR.

**Review by jsancio — COMMENTED** [review]
@cmccabe Can you review this when you get a chance?

File: raft/src/main/java/org/apache/kafka/raft/FollowerState.java [hunk 2]
```diff
@@ -137,11 +137,11 @@ public boolean updateHighWatermark(OptionalLong highWatermark) {
...
         if (fetchingSnapshot.isPresent()) {
             fetchingSnapshot.get().close();
         }
```
Comment (mumrah) [inline raft/src/main/java/org/apache/kafka/raft/FollowerState.java:143]: Ok, so previously we were closing whatever snapshot was passed in to `setFetchingSnapshot?`

**Review by mumrah — APPROVED** [review]
Thanks @jsancio. Looks like a straightforward fix. 🚢

**Pull request merged**